File: packages/es-markdown-core/src/model/specimen.ts
import type { Markup } from "./markup.js";
import type { Scope } from "../values.js";

/** A document-owned specimen definition; its displayed number is derived. */
export interface Specimen {
    readonly scope: Scope;
    /** The authored label, or null for an anonymous definition. */
    readonly id: string | null;
    /** An explicit effective counter reset, or null for continuation. */
    readonly start: number | null;
    readonly content: readonly Markup[];
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Specimen.kt
package com.nouprax.markdown.core

/** A document-owned Markup definition with authored label and counter-reset facts. */
public class Specimen internal constructor(
    /** The authored label, or null for an anonymous definition. */
    public val id: String?,
    /** An explicit counter reset, or null when numbering continues. */
    public val start: Long?,
    public val content: kotlin.collections.List<Markup>,
    override val scope: Scope,
    override val anchor: String? = null,
    override val attributes: Attributes = Attributes.empty,
) : Markup {
    override fun <Result> accept(
        visitor: Visitor<Result>,
        phase: MarkupWalkPhase,
    ): Result = visitor.visit(this, phase)
}
